modules/globebrowsing/cache/lrucache.h

```diff
@@ -45,17 +45,39 @@ class LRUCache {
     LRUCache(size_t size);
 
     void put(const KeyType& key, const ValueType& value);
+    std::vector<std::pair<KeyType, ValueType>> putAndFetchPopped(const KeyType& key, const ValueType& value);
     void clear();
     bool exist(const KeyType& key) const;
+    /**
+     * If value exists, the value is bumped to the front of the queue.
+     * \returns true if value of this key exists.
+     */
+    bool touch(const KeyType& key);
+    bool isEmpty() const;
     ValueType get(const KeyType& key);
+
+    /**
+     * Pops the front of the queue.
+     */
+    ValueType popMRU();
+    
+    /**
+     * Pops the back of the queue.
+     */
+    ValueType popLRU();
     size_t size() const;
 
 private:
+    void putWithoutCleaning(const KeyType& key, const ValueType& value);
     void clean();
+    std::vector<std::pair<KeyType, ValueType>> cleanAndFetchPopped();
+
+    using Item = std::pair<KeyType, ValueType>;
+    using Items = std::list<Item>;
```

Comment:
I might have a look at comparing the performance of vector and list in for this use case later on.